### PR TITLE
Persist session scenario

### PR DESCRIPTION
**Pull request by k32**
TODO:
1. More accurate throughput metrics
2. Handle dropped messages (currently the scenario just hangs)

### CODE REVIEW

File: src/scenarios/emqttb_scenario_persistent_session.erl [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+              , type => non_neg_integer()
+              , cli_operand => "size"
+              , cli_short => $s
+              , default => 256
+              }}
+        , pubinterval =>
+            {[value, cli_param],
+             #{ oneliner => "Message publishing interval"
+              , type => emqttb:interval()
+              , default_ref => [interval]
```
Comment (thalesmg) [inline src/scenarios/emqttb_scenario_persistent_session.erl:94]: Is it possible to know which `interval` this refers to?  Is it the interval in the root model?

Also, is `default_ref` to use a default value set in another mnode?

File: src/scenarios/emqttb_scenario_persistent_session.erl [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+            {[value, cli_param],
+             #{ oneliner => "Suffix of the topic to publish to"
+              , type => binary()
+              , default => <<"%h/%n">>
+              , cli_operand => "topic"
+              , cli_short => $t
+              }}
+        , pub_time =>
+            {[value, cli_param],
+             #{ oneliner => "Suffix of the topic to publish to"
```
Comment (thalesmg) [inline src/scenarios/emqttb_scenario_persistent_session.erl:131]:
```suggestion
             #{ oneliner => "Period of time while publishing will last (ms)"
```
?

File: src/scenarios/emqttb_scenario_persistent_session.erl [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+              , type => non_neg_integer()
+              , default => 5_0000
```
Comment (thalesmg) [inline src/scenarios/emqttb_scenario_persistent_session.erl:133]:
```suggestion
              , default => 50_000
```

File: src/scenarios/emqttb_scenario_persistent_session.erl (hunk 1 above)
Comment (k32) [reply src/scenarios/emqttb_scenario_persistent_session.erl:94]: Yes, default_ref takes the default value from another value. Here `[interval]` is the global key.

File: src/scenarios/emqttb_scenario_persistent_session.erl (hunk 3 above)
Comment (ieQu1) [reply src/scenarios/emqttb_scenario_persistent_session.erl:133]: I found that shorter bursts of traffic lead to more stable tests, when too many messages are queued up, EMQX tends to start dropping messages.

Comment (k32) [reply src/scenarios/emqttb_scenario_persistent_session.erl:133]: I found that shorter bursts of traffic lead to more stable tests, when too many messages are queued up, EMQX tends to start dropping messages.

**Pull request merged**